src/Raven.Server/Documents/Handlers/Debugging/NodeDebugHandler.cs

```diff
@@ -103,79 +107,152 @@ await using (var writer = new AsyncBlittableJsonTextWriterForDebug(context, Serv
                 writer.WritePropertyName("Result");
 
                 writer.WriteStartArray();
+
                 while (tasks.Count > 0)
                 {
                     var task = await Task.WhenAny(tasks);
                     tasks.Remove(task);
                     context.Write(writer, task.Result.ToJson());
+
+                    if (setStatusCodeOnError && task.Result.HasErrors)
+                        HttpContext.Response.StatusCode = (int)HttpStatusCode.ServiceUnavailable;
```

Comment:
status code, headers, etc should be set before anything is written to the response. At this point you do not have guarantee that response have not started, because few lines up you are writing to it.

Reply:
done